src/keyboard-layout-manager-linux.cc

```diff
@@ -46,9 +60,17 @@ KeyboardLayoutManager::KeyboardLayoutManager(v8::Isolate *isolate, Nan::Callback
       XNFocusWindow, window, NULL
     );
   }
+
+#if NODE_MAJOR_VERSION >= 10
+  node::AddEnvironmentCleanupHook(
+    isolate,
+    CleanContext,
+    const_cast<void*>(static_cast<const void*>(this)));
+#endif
 }
 
 KeyboardLayoutManager::~KeyboardLayoutManager() {
+#if NODE_MAJOR_VERSION < 10
```

Comment:
same comment as in the destructor of the  .mm file

Reply:
Code refactores